catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -611,15 +624,21 @@ protected UpdateResponse patchMetacards(List<MetacardChanges> metacardChanges)
             for (AttributeChange attributeChange : changeset.getAttributes()) {
                 for (String id : changeset.getIds()) {
                     List<String> values = attributeChange.getValues();
-                    Metacard result = results.get(id)
-                            .getMetacard();
+
+                    Result result = results.get(id);
+                    if (result == null) {
+                        LOGGER.debug("Metacard with id {} not found in result set", id);
+                        continue;
+                    }
+
+                    Metacard metacard = result.getMetacard();
```

Comment:
So this is changing the behavior quite a bit and there's a couple options depending on what route we want to go. It's one of those cases where we have to decide if we want to allow partial successes or not.

1) If we leave it like this (so it silently skips), we need to update the method that calls this to remove the things that we didn't actually update from the response we give back. 
2) Make it so we throw an exception here and it fails the transaction completely. (and make sure we return the proper error message to the user).

Personally I would suggest throwing an exception instead of continuing because its less ambiguous. Partial updates can be very confusing to a user (especially since we don't have anything in place to let the user know which things succeeded and failed).